kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/PortConflictDetector.java

```diff
@@ -8,242 +8,114 @@
 
 import java.util.Collection;
 import java.util.Comparator;
-import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
-import java.util.Map;
 import java.util.Optional;
-import java.util.Set;
-import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import io.kroxylicious.proxy.internal.net.EndpointListener;
 import io.kroxylicious.proxy.model.VirtualClusterModel;
 import io.kroxylicious.proxy.service.HostPort;
 
+import edu.umd.cs.findbugs.annotations.NonNull;
+
 /**
  * Detects potential for port conflicts arising between virtual cluster configurations.
  */
 @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
 public class PortConflictDetector {
 
-    private static final Optional<String> ANY_INTERFACE = Optional.empty();
     private static final String ANY_STRING = "<any>";
 
     private enum BindingScope {
         SHARED,
         EXCLUSIVE
     }
 
-    /**
-     * Validates the configuration throwing an exception if a conflict is detected.
-     *
-     * @param virtualClusterModelMap map of virtual clusters.
-     * @param otherExclusivePort an optional exclusive port that should conflict with virtual cluster ports
-     */
-    public void validate(Collection<VirtualClusterModel> virtualClusterModelMap, Optional<HostPort> otherExclusivePort) {
-
-        Set<String> seenVirtualClusters = new HashSet<>();
-
-        Map<Optional<String>, Set<Integer>> inUseExclusivePorts = new HashMap<>();
-        Map<Optional<String>, Map<Integer, Boolean>> inUseSharedPorts = new HashMap<>();
+    private record BindHost(Optional<String> host) {
```

Comment:
naming: 'host' isn't quite right, you bind to an interface.  BindHost -> BindAddress perhaps.